### ISSUE TITLE
Navigation problem when ClickShortcut is present in LegacyWrapper

### ISSUE THREAD

**Issue by paodb** [vaadin/multiplatform-runtime#115]
### Description
Navigation seems to break when trying to navigate to a Flow View containing a LegacyWrapper of a Vaadin 7 component. The issue seems to be caused when adding a ClickShortcut to a Vaadin 7 Button that is inside a Legacy Wrapper in a Vaadin Flow View.  If I remove that ClickShortcut, the navigation works okay. See attached image of the problem:

![mpr_navigation1](https://user-images.githubusercontent.com/30666649/181245097-0ca04d69-98a3-47db-a97d-b16416351bec.gif)

### Minimal reproducible example

```java
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.Route;

@Route(value = "navigate")
public class NavigatePage extends VerticalLayout {

    public NavigatePage() {
        Button button = new Button("Navigate", event -> {
            event.getSource().getUI().ifPresent(ui -> ui.navigate("button"));
        });

        add(button);
    }
}
```

```java
import com.vaadin.event.ShortcutAction;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.mpr.LegacyWrapper;
import com.vaadin.ui.Button;

@Route(value = "button")
public class ButtonPage extends VerticalLayout {

    public ButtonPage() {
        Button button = new Button("Vaadin 7 Button");
        button.setClickShortcut(ShortcutAction.KeyCode.ENTER); // comment to make it work

        add(new LegacyWrapper(button));
    }
}
```

Complete example here: [mpr_navigation_sources.zip](https://github.com/vaadin/multiplatform-runtime/files/9198734/mpr_navigation_sources.zip)

### Versions
- Vaadin Framework: 7.7.32
- Flow: 23.1.3
- MPR: 6.0.1
- Java JRE / JDK: 11

**Comment by MarcinVaadin** [comment]
Investigation results:

Why going _root -> Button_ or _navigate -> Button_ does not brake the location? Because:
1. Framework UI is being lazely loaded when opening view with v7 component
2. Current location is stored in v7 `Page` object
3. Two UIDL updates are returned, one for Flow (no dirty v7 components for state update yet) and one after Framework initialization

And why going back and forth is breaking location:
1. Adding clickShortcut is causing `MprUI` to be marked as dirty
2. Dirty visible components state is added to UIDL to force component repaint
5. `MprUI` state contains `location` from `Page` instance (see point 2 from list above)
6. Vaadin 23 `UidlRequestHandler.removeOffendingMprHashFragment` is looking for `location` and forces unnecessary history state push (with previously stored improper location) - (introduced in https://github.com/vaadin/flow/pull/7693)

**Comment by mshabarov** [comment]
Indeed, this is a regression after applying a patch for fixing https://github.com/vaadin/flow/issues/7540 .

`UidlRequestHandler::removeOffendingMprHashFragment` adds an extra `http://localhost:8080` to the history, through this method:
https://github.com/vaadin/flow/blob/020ee677e59d3b0f411232e402c55a68f34924f2/flow-server/src/main/java/com/vaadin/flow/server/communication/UidlRequestHandler.java#L309

The navigation works well if the server-side bootstrapping is turned on through the parameter `vaadin.useDeprecatedV14Bootstrapping=true` in `application.properties`. In server-side bootstrapping mode the fragment manipulations are disabled. This can be used as a workaround until the issue is fixed.

**Comment by kshashov** [comment]
We tried turning `useDeprecatedV14Bootstrapping` on but the issue was still reproducible in a slightly different way
![mpr_navigation2](https://user-images.githubusercontent.com/3830719/184921407-991cd319-3e01-46bd-9e50-95d040b6d212.gif)

**Comment by mshabarov** [comment]
@kshashov thanks, indeed, this also doesn't fix the problem. Looks like a completely different issue in Flow https://github.com/vaadin/flow/issues/14323 .

**Comment by Tostino** [comment]
Looking for guidance on when the next hotfix release will be that will include this, or the best way to incorporate this just this fix into our build, as we need to get our software released very soon and are looking for the path that gives us the least resistance here.

**Comment by mshabarov** [comment]
@Tostino which version of Vaadin do you use? Is it V23.1.x ? Since the fix is on the Flow side, I can release the corresponding version of Flow with this patch tomorrow morning and you can override it in your project. Platform release comes later on according to a normal schedule, so once it's ready you would remove the override and just upgrade the platform version. Does it sound fine for you?

**Comment by Tostino** [comment]
We made the original report, so we are on 23.1.3 right now. That seems like it'll work just fine, we can just upgrade the flow version. 

Thanks for the help here.

**Comment by mshabarov** [comment]
@Tostino please try a new Flow 23.1.6 with this patch:

```
    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.vaadin</groupId>
                <artifactId>flow-bom</artifactId>
                <version>23.1.6</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
            <dependency>
                <groupId>com.vaadin</groupId>
                <artifactId>vaadin-bom</artifactId>
                <type>pom</type>
                <scope>import</scope>
                <version>23.1.6</version>
            </dependency>
        </dependencies>
    </dependencyManagement>
```

Will be included into next 23.1.x platform release.
